src/config/ServerConfig.ts

```diff
@@ -0,0 +1,11 @@
+import deepFreeze from 'deep-freeze';
+import { store } from '../store/ConfigStore';
+import { KialiAppState, ServerConfig } from '../store/Store';
+import { PersistPartial } from 'redux-persist';
+
+// It's not great to access the store directly for convenience but the alternative is
+// a huge code ripple just to access some server config. better to just have this one utility.
+export const serverConfig = (): ServerConfig => {
```

Comment:
I'll leave this to the redux gurus if there's a better approach :-)

Reply:
I think the alternate approach would be to have every user of the serverConfig be containerized and I just think that's overkill because in several places we need it in "utility" code.  I actually went down this path and it was getting out of hand.   I containerized the actual classes that need to respond to a change for re-render (MetricsDuration, GraphRefresh, OverviewToolbar).

Reply:
What I understand is that the `serverConfig`, once set after logging-in or after validating the stored token (because a browser refresh), the values will never change.

Personally, I would have avoided redux. Since the values never mutate once set, *I think* the server config is not really part of *the state of the app*, but just global values. Thus, redux looks overkill for this use-case.
 
Instead, what I'd have done is to create a separate module to store these values as plain global variables. Something similar how the `history` object is handled in https://github.com/kiali/kiali-ui/blob/929598bd0f324219e33e1a9bdbb2004b3de7abdc/src/app/History.ts. On that separate module you could also have placed all utility functions to retrieve the "valid durations" and any other needed helper functions. Then, you could have just imported that new module where needed and read the global variables leaving redux aside and avoiding the burden of reducers, actions, component containerization, etc. 

Then, your `completeLogin` method, instead of dispatching a `setServerConfig` action, would had simply set the global variables.

But you have already done all that redux burden, is working as is and it's quite OK. So this is only a suggestion about considering keeping redux aside the next time global variables can do the work.

Reply:
> I have minor comments for your consideration, but nothing that should block this.

@israel-hdez You are funny, as this is a major comment :)   But you are also correct, the redux 
approach is not fantastic, and it wasn't my first approach.   You are right that the values will not change between login or a credential check.  I did not start with adding serverConfig to redux but I did it for two reasons.  First, by having it in redux we can better re-establish the state surrounding a problem.  The second reason was more critical.  When navigator dispatches LoginThunkActions.checkCredentials (e.g. on an F5) the logic does not block the rendering, so what happens is the page starts rendering before the serverConfig is set, and that's a major problem as there is nothing set when the page needs to reference serverConfig().

I am actually quite OK with an approach that does not use redux if the call to serverConfig() is guaranteed to return the actual serverConfig (or thrown an error).  This could possibly be done  by somehow making checkCredentials block (pehaps convert it from a thunk?), or if a call to serverConfig() fetched it if needed, setting a global the first time it is called (note that it would need an authorization token to do so).  In the second approach login/checkCredentials would not actually need to set serverConfig, and only checkCredentials would need call it, just as an auth check.

I am OK if you want to submit a PR to this PR with a different approach.

Reply:
I'm not sure if it's easy or not to have in our current implementation, but `serverConfig()` could also just return a promise, so that anything that calls it would have to deal with the result asynchronously.

Reply:
@israel-hdez After talking to you offline the decision is to leave the redux solution in place as it is working and an alternative approach could be investigated in a later refactoring.  @jotak, a promise approach may work, the component would have to defer rendering until the response was received.  Also, we still need to ensure we have an auth token before we can make the call.